errors.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package goacme
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+// ErrorType is used to define a set of errors predefined by the ACME spec.
+type ErrorType string
+
+const (
+	ErrBadCSR       = "urn:acme:error:badCSR"         // The CSR is unacceptable (e.g., due to a short key)
```

Comment:
I am not a Golang expert, but my understanding is that the proper solution for error reporting is to define global vars in the module that actually *are* of struct type of error so that you can do plain comparison in the client. Strings still require string-based comparison if I understand this correctly